pybigquery/sqlalchemy_bigquery.py

```diff
@@ -294,53 +302,67 @@ def _add_default_dataset_to_job_config(job_config, project_id, dataset_id):
+
+        if credentials:
+            scopes = (
                 'https://www.googleapis.com/auth/bigquery',
                 'https://www.googleapis.com/auth/cloud-platform',
                 'https://www.googleapis.com/auth/drive'
             )
-        credentials = credentials.with_scopes(scopes)
-
-        self._add_default_dataset_to_job_config(default_query_job_config, project_id, self.dataset_id)
+            credentials = credentials.with_scopes(scopes)
 
-        return bigquery.Client(
-                project=project_id,
-                credentials=credentials,
-                location=self.location,
-                default_query_job_config=default_query_job_config,
-            )
+        return credentials
 
     def create_connect_args(self, url):
-        project_id, location, dataset_id, arraysize, credentials_path, default_query_job_config = parse_url(url)
+        project_id, location, dataset_id, arraysize, credentials_path, use_bqstorage_api, default_query_job_config = parse_url(url)
 
         self.arraysize = self.arraysize or arraysize
         self.location = location or self.location
         self.credentials_path = credentials_path or self.credentials_path
         self.dataset_id = dataset_id
 
-        if self.credentials_path:
-            credentials = service_account.Credentials.from_service_account_file(self.credentials_path)
-            client = self._create_client_from_credentials(credentials, default_query_job_config, project_id)
+        credentials = self.create_credentials(self.credentials_path, self.credentials_info)
 
-        elif self.credentials_info:
-            credentials = service_account.Credentials.from_service_account_info(self.credentials_info)
-            client = self._create_client_from_credentials(credentials, default_query_job_config, project_id)
+        # Use the credentials project as a default if no project was specefied
+        if (credentials is not None) and (project_id is None):
+            project_id = credentials.project_id
 
-        else:
-            self._add_default_dataset_to_job_config(default_query_job_config, project_id, dataset_id)
+        self._add_default_dataset_to_job_config(default_query_job_config, project_id, dataset_id)
 
-            client = bigquery.Client(
-                project=project_id,
-                location=self.location,
-                default_query_job_config=default_query_job_config
-            )
+        client = bigquery.Client(
+            project=project_id,
+            credentials=credentials,
+            location=self.location,
+            default_query_job_config=default_query_job_config
+        )
+        
+        clients = [client]
+
+        if use_bqstorage_api:
+            if parse_version(bigquery.__version__) >= Version("1.26.0"):
+                try:
+                    storage_client = bigquery_storage_v1.BigQueryReadClient(credentials=credentials)
+                    clients.append(storage_client)
```

Comment:
Since `bqstorage_client` is an optional argument to [`connect()`](https://googleapis.dev/python/bigquery/latest/dbapi.html#google.cloud.bigquery.dbapi.connect), we should probably be using the dictionary for this rather than appending to the list of positional arguments. (In fact, maybe we should do the same for the BigQuery client object, too)